### PR TITLE
Feature Implementation - Recipe app

### PR DESCRIPTION
**Pull request by AndreaM2429**
Hello there!
@PabloBona and @AndreaM2429 here!

## 📋 Description
This pull request implements the required features outlined in the project specifications. The changes satisfy all the requirements, including the correct use of Gitflow, adherence to Ruby best practices, and proper documentation. The implemented features include login and registration pages, food list functionality, recipe management, public recipe display, recipe details, shopping list generation, and navigation menus.

## 🛠️ Milestones 🛠️
- Implemented login and registration pages using Devise for authentication.
- Created a food list that displays foods added by the logged-in user, including quantities, adhering to wireframe layouts.
- Implemented recipe list functionality for users, allowing viewing and deleting their recipes, following the provided wireframe.
- Developed a public recipe list, sorting recipes by newest, and enabling users to view recipe details.
- Created recipe details view, including the option to toggle recipes between public and private status.
- Implemented a shopping list view that displays missing food items needed for all user recipes, counting total items and prices.
- Created a navigation menu for seamless navigation between pages.
- Implemented unit and integration tests using RSpec and Capybara to ensure functionality and prevent regressions.

## Technical Details
- **Gitflow:** Followed the Gitflow branching model for feature development, ensuring a clear version control history.
- **Authentication:** Implemented authentication using Devise for secure user login and registration.
- **Testing:** Utilized RSpec and Capybara for comprehensive unit and integration testing, ensuring robust code quality.
- **Performance:** Ensured no N+1 queries are present, optimizing database interactions for efficiency.
- **Documentation:** Documented the codebase professionally, providing insights into implementation details and logic flow.

## 👨‍💻 Testing 
- Unit tests cover individual components and models, verifying their functionality in isolation.
- Integration tests ensure seamless interaction between different modules, guaranteeing feature integration correctness.

## Additional Notes
- The implemented features strictly adhere to the wireframe layouts, incorporating personalized design elements such as colors, typographies, and spacings.
- Linter errors have been eliminated, and the codebase follows the prescribed Ruby best practices.

## Screenshots
image-1
image-2
image-3
image-4

## 📝 Checklist
- [x] Implemented all specified features and functionalities.
- [x] Adhered to Gitflow branching model.
- [x] Utilized Devise for authentication.
- [x] Incorporated RSpec and Capybara for unit and integration testing.
- [x] Ensured no N+1 queries in database interactions.
- [x] Followed the provided wireframes and personalized design elements.
- [x] Eliminated linter errors and followed Ruby best practices.
- [x] Properly documented the codebase.
- [x] Created a navigation menu for a seamless user experience.

### Reviewer 
Thank you for taking the time and effort to review this PR ⭐!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,123 @@
...
+
+Contributions, issues, and feature requests are welcome! 
+Feel free to check the issues [page](https://github.com/PabloBona/recipe-app) and contribute to the project.
+
+## ⭐️ Show Your Support <a name="support"></a>
+If you find this project helpful, consider giving it a ⭐️ to show your support!
+
+## 📜 License <a name="license"></a>
+This project is [MIT](https://github.com/PabloBona/Blog-app/recipe-app/dev/MIT.md) licensed.
+
```
Comment (Olamarx) [inline README.md]: - [x] This link goes to a 404 page, I will encourage you to fix it.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- All 60 test cases passed ✔️
- Meaningful commit ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (AndreaM2429) [reply README.md]: Hi @Olamarx, done.
Thanks for the observation.

Comment (peterrobert) [reply README.md]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @AndreaM2429, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work making the change requested by the reviewer. ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**